exporters/otlp/test/otlp_http_exporter_test.cc

```diff
@@ -0,0 +1,323 @@
+  resource_attributes["vec_string_value"] = std::vector<std::string>{"vector", "string"};
+  auto resource = opentelemetry::sdk::resource::Resource::Create(resource_attributes);
+
+  auto processor_opts                  = sdk::trace::BatchSpanProcessorOptions();
+  processor_opts.max_export_batch_size = 5;
+  processor_opts.max_queue_size        = 5;
+  processor_opts.schedule_delay_millis = std::chrono::milliseconds(256);
+
+  auto processor = std::unique_ptr<sdk::trace::SpanProcessor>(
+      new sdk::trace::BatchSpanProcessor(std::move(exporter), processor_opts));
+  auto provider = nostd::shared_ptr<trace::TracerProvider>(
+      new sdk::trace::TracerProvider(std::move(processor), resource));
+
+  std::string report_trace_id;
+  {
+    uint8_t trace_id_binary[opentelemetry::trace::TraceId::kSize] = {0};
+    auto tracer                                                   = provider->GetTracer("test");
+    auto parent_span = tracer->StartSpan("Test parent span");
+
+    opentelemetry::trace::StartSpanOptions child_span_opts = {};
+    child_span_opts.parent                                 = parent_span->GetContext();
+
+    auto child_span = tracer->StartSpan("Test child span", child_span_opts);
+    child_span->End();
+    parent_span->End();
+
+    child_span_opts.parent.trace_id().CopyBytesTo(MakeSpan(trace_id_binary));
+    report_trace_id.assign(reinterpret_cast<char *>(trace_id_binary), sizeof(trace_id_binary));
+  }
+
+  ASSERT_TRUE(waitForRequests(2, old_count + 1));
+
+  auto received_trace_id = received_requests_binary_.back()
+                               .resource_spans(0)
+                               .instrumentation_library_spans(0)
+                               .spans(0)
+                               .trace_id();
+  EXPECT_EQ(received_trace_id, report_trace_id);
+}
+
+// Test exporter configuration options
+TEST_F(OtlpHttpExporterTestPeer, ConfigTest)
+{
+  OtlpHttpExporterOptions opts;
+  opts.url = "http://localhost:45455/v1/traces";
+  std::unique_ptr<OtlpHttpExporter> exporter(new OtlpHttpExporter(opts));
+  EXPECT_EQ(GetOptions(exporter).url, "http://localhost:45455/v1/traces");
+}
+
+// Test exporter configuration options with use_json_name
+TEST_F(OtlpHttpExporterTestPeer, ConfigUseJsonNameTest)
+{
+  OtlpHttpExporterOptions opts;
+  opts.use_json_name = true;
+  std::unique_ptr<OtlpHttpExporter> exporter(new OtlpHttpExporter(opts));
+  EXPECT_EQ(GetOptions(exporter).use_json_name, true);
+}
+
+// Test exporter configuration options with json_bytes_mapping=JsonBytesMappingKind::kHex
+TEST_F(OtlpHttpExporterTestPeer, ConfigJsonBytesMappingTest)
+{
+  OtlpHttpExporterOptions opts;
+  opts.json_bytes_mapping = JsonBytesMappingKind::kHex;
+  std::unique_ptr<OtlpHttpExporter> exporter(new OtlpHttpExporter(opts));
+  EXPECT_EQ(GetOptions(exporter).json_bytes_mapping, JsonBytesMappingKind::kHex);
+}
+
+}  // namespace otlp
+}  // namespace exporter
+OPENTELEMETRY_END_NAMESPACE
+#endif
```

Comment:
Thank you for adding these real tests using the inbuilt HTTP server. We should be doing something similar for Zipkin too. I will add an issue for same.